lib/open_api/serializer.rb

```diff
@@ -1,14 +1,29 @@
 module OpenApi
   module Serializer
     def as_json(arg = {})
-      return super unless kind_of?(ApplicationRecord)
-
-      super.tap do |hash|
-        hash.each_key do |attr_key|
-          hash[attr_key] = hash[attr_key].iso8601 if hash[attr_key].kind_of?(Time)
-          hash[attr_key] = hash[attr_key].to_s if attr_key.ends_with?("_id") || attr_key == "id"
-        end
+      previous = super
+      encrypted_columns_set = (self.class.try(:encrypted_columns) || []).to_set
```

Comment:
@syncrou can we break this into smaller functions
 * get_path
 * get_object_def_from_path
 * filter

Reply:
@mkanoor I refactored as much as I could.  Left the `get_` prefix out of the names to keep rubocop happy.